controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -462,16 +496,34 @@ public void listReaderGroups(final String scopeName, final SecurityContext secur
     public void listScopes(final SecurityContext securityContext, final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "listScopes");
 
+        Principal principal;
+        try {
+            principal = authenticate();
+            authorize("/", principal, READ);
+        } catch (AuthException e) {
+            log.warn("Get scopes failed due to authentication failure.");
+            asyncResponse.resume(Response.status(Status.fromStatusCode(e.getResponseCode())).build());
+            LoggerHelpers.traceLeave(log, "listScopes", traceId);
+            return;
+        }
+
         controllerService.listScopes()
-                .thenApply(scopesList -> {
-                    ScopesList scopes = new ScopesList();
-                    scopesList.forEach(scope -> scopes.addScopesItem(new ScopeProperty().scopeName(scope)));
-                    return Response.status(Status.OK).entity(scopes).build();
-                }).exceptionally(exception -> {
-                        log.warn("listScopes failed with exception: " + exception);
-                        return Response.status(Status.INTERNAL_SERVER_ERROR).build();
-                }).thenApply(asyncResponse::resume)
-                .thenAccept(x ->  LoggerHelpers.traceLeave(log, "listScopes", traceId));
+                         .thenApply(scopesList -> {
+                             ScopesList scopes = new ScopesList();
+                             scopesList.forEach(scope -> {
+                                 try {
+                                     authorize(scope, principal, READ);
+                                     scopes.addScopesItem(new ScopeProperty().scopeName(scope));
+                                 } catch (AuthException e) {
+                                     log.warn("Not adding {} to list because authentication exception {}", scope, e);
+                                 }
+                             });
+                             return Response.status(Status.OK).entity(scopes).build();
+                         }).exceptionally(exception -> {
```

Comment:
Please rework this method's indentation. The next line starts before the upper line which makes this very hard to read

Reply:
Fixed.